### PR TITLE
bug(parsing): inject pageSlug with Githubpublish.js

### PR DESCRIPTION
**Pull request by langdonholmes**
This is messy.

Unfortunately, I can't find a way to locate any information about the parent page in a chunk's lifecycle event.

I partially solved this by querying the Strapi database to find the parent page, but this only works after the database entries have been created.

For now, this means that REPLs will have a filler pageSlug until the chunk is updated.

I honestly can't think of any workaround here that doesn't bypass Strapi lifecycles altogether, but this is kind of functional and might hold us over until we can figure something out.

### CODE REVIEW

**Comment by langdonholmes** [comment]
Oh...

Just realized we can solve this with the GitHub Publish script. I kind of hate splitting up the logic into multiple places, but in this case, I don't think there's a better option.

I'll just rewrite this to leave a placeholder value as the pageslug. Then we can replace that with the real pageslug in the GitHub Publish script.

**Comment by theannachen** [comment]
Do you think it's wise for me to add the GithubPublish update as a commit in this PR

**Comment by langdonholmes** [comment]
I need to clean up most of the code I wrote and revert the turndown service to a simple function (I made it a factory to support my first attempt at this).

If you want to add a change to the GitHub Publish script, that would be great. Let's assume the slug placeholder will be `__temp_slug__`. This PR is fine.

I should have a few minutes to work on this tomorrow.

**Comment by theannachen** [comment]
Sounds good! I will add a change soon

File: githubpublish.js [hunk 1]
```diff
@@ -131,15 +131,15 @@ async function entryPages(textData, startingPath) {
...
         if (curChunk.MDX != null) {
           stream.write(
-            curChunk.MDX.replace(/[\u200B-\u200D\uFEFF\u00A0]/g, "").replace(
-              /(<br\s*\/?>\s*)+/g,
-              "\n\n"
-            )
+            curChunk.MDX.replaceAll(/[\u200B-\u200D\uFEFF\u00A0]/g, "")
+              .replaceAll(/(<br\s*\/?>\s*)+/g, "\n\n")
+              .replaceAll("__temp_slug__", pageData["Slug"])
+              .replaceAll("test-page-no-chunks", pageData["Slug"]),
```
Comment (langdonholmes) [inline githubpublish.js:142]: Need a comment here.

File: githubpublish.js [hunk 2]
```diff
@@ -131,15 +131,15 @@ async function entryPages(textData, startingPath) {
...
           );
         }
         if (curChunk.MDX != null) {
           stream.write(
-            curChunk.MDX.replace(/[\u200B-\u200D\uFEFF\u00A0]/g, "").replace(
-              /(<br\s*\/?>\s*)+/g,
-              "\n\n"
-            )
+            curChunk.MDX.replaceAll(/[\u200B-\u200D\uFEFF\u00A0]/g, "")
+              .replaceAll(/(<br\s*\/?>\s*)+/g, "\n\n")
```
Comment (langdonholmes) [inline githubpublish.js:140]: We are replacing "br" elements with "<br/>" in `mdx-service.js`. Why are we replacing them with something else here?

This logic is convoluted. Let's just replace "br" elements with "\n\n" in mdx-service directly.

File: githubpublish.js [hunk 3]
```diff
@@ -131,15 +131,15 @@ async function entryPages(textData, startingPath) {
...
+            `<h2 className="sr-only" id="${chunkSlug}">${curChunk.Header}</h2>\n\n`,
           );
         }
         if (curChunk.MDX != null) {
           stream.write(
-            curChunk.MDX.replace(/[\u200B-\u200D\uFEFF\u00A0]/g, "").replace(
-              /(<br\s*\/?>\s*)+/g,
-              "\n\n"
-            )
+            curChunk.MDX.replaceAll(/[\u200B-\u200D\uFEFF\u00A0]/g, "")
```
Comment (langdonholmes) [inline githubpublish.js:139]: Please add a comment here. I had to Google what this RegEx does.

**Review by langdonholmes — COMMENTED** [review]
Looks good. Please just add some inline comments where I indicated and remove the linebreak replacement from Githubpublish.js. We should just perform the replacement that we want in mdx-parsing.js to begin with.

**Comment by langdonholmes** [comment]
```js
turndownService.addRule("LineBreaks", {
  filter: "br",
  replacement: function () {
    return "<br/>";
  },
});
```

This rule still exists in mdx-parsing, so newly created HTML embeds will also have "<br/>".

Can we replace the above rule with the following?

```js
turndownService.addRule("LineBreaks", {
  filter: "br",
  replacement: function () {
    return "\n\n";
  },
});
```

**Review by theannachen — APPROVED** [review]

**Pull request merged**